config/CMakeLists.txt

```diff
@@ -121,6 +121,32 @@ configure_file (
   )
 
 # generate version definitions
+if(NVIM_VERSION_MEDIUM)
+  message(STATUS "Using version from NVIM_VERSION_MEDIUM: ${NVIM_VERSION_MEDIUM}")
+elseif(EXISTS ${PROJECT_SOURCE_DIR}/.git)
+  find_program(GIT_EXECUTABLE git)
+  if(GIT_EXECUTABLE)
+    # Get current version.
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} describe --dirty
+      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+      OUTPUT_VARIABLE NVIM_VERSION_MEDIUM
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    message(STATUS "Using version from git: ${NVIM_VERSION_MEDIUM}")
+
+    # Create a update_version_stamp target to update the version during build.
+    add_custom_target(update_version_stamp ALL
+      COMMAND ${PROJECT_SOURCE_DIR}/scripts/build/update_version_stamp.sh
+        ${CMAKE_BINARY_DIR}/.version_stamp
+        ${CMAKE_BINARY_DIR}/config/auto/versiondef.h
+      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+      BYPRODUCTS ${CMAKE_BINARY_DIR}/config/auto/versiondef.h)
```

Comment:
CMake 2.8 does not support BYPRODUCTS.
Instead of working around it, I think we should rather bump the minimum version ourself.
There's an effort being done in this regard for libuv already (https://github.com/libuv/libuv/pull/2495), and the bundled libuv would not work with CMake 2.8 then anymore anyway.